src/main/java/filter/expression/Qualifier.java

```diff
@@ -136,28 +133,74 @@ public static boolean process(IModel model, FilterExpression expr, TurboIssue is
+                .collect(Collectors.toList()));
+
+        // Special case: if there is only one open milestone, it should be included regardless of whether
+        // it has due date or not.
+        // In this case, if it is not included already (the open milestone does not
+        // have due date), add it to the milestone list.
+        List<TurboMilestone> openMilestones = TurboMilestone.getOpenMilestones(milestonesOfRepos);
+        if (openMilestones.size() == 1) {
+            TurboMilestone openMilestone = openMilestones.get(0);
+            boolean hasDueDate = openMilestone.getDueDate().isPresent();
+            if (!hasDueDate) {
+                milestones.add(openMilestone);
+            }
         }
 
-        int currentIndex = 0;
+        return milestones;
+    }
 
-        for (TurboMilestone checker : allMilestones) {
-            boolean overdue = checker.getDueDate().isPresent() &&
-                    checker.getDueDate().get().isBefore(LocalDate.now());
-            boolean relevant = !(overdue && checker.getOpenIssues() == 0);
+    /**
+     * Returns a map where the key denotes offset from "current" milestone.
+     * i.e. 0 for "current" milestone, -1 for "current-1" milestone, etc.
+     */
+    private static Map<Integer, TurboMilestone> getMilestoneAliasIndex(List<TurboMilestone> milestones) {
+        List<TurboMilestone> sortedMilestones = TurboMilestone.getSortedMilestonesByDueDate(milestones);
+        Optional<Integer> currentMilestoneIndex = getCurrentMilestoneIndex(sortedMilestones);
 
-            if (checker.isOpen() && relevant) {
-                return Optional.of(currentIndex);
-            }
+        return IntStream
+                .range(0, sortedMilestones.size())
+                .boxed()
+                .collect(Collectors.toMap(
+                        i -> i - currentMilestoneIndex.get(),
+                        i -> sortedMilestones.get(i)));
+    }
 
-            currentIndex++;
+    private static Optional<Integer> getCurrentMilestoneIndex(List<TurboMilestone> milestones) {
+        if (milestones.isEmpty()) {
+            return Optional.empty();
         }
 
-        // if no open milestone, set current as one after last milestone
-        // - this means that no such milestone, which will return no issue
-        return Optional.of(allMilestones.size());
+        long openMilestonesCount = milestones.stream().filter(TurboMilestone::isOpen).count();
+        if (openMilestonesCount == 1) {
+            return IntStream
+                    .range(0, milestones.size())
+                    .boxed()
+                    .filter(i -> milestones.get(i).isOpen())
+                    .findFirst();
+        } else {
+            // Look for the first milestone in the (sorted) list that is both open and ongoing.
+            int currentIndex = 0;
+            for (TurboMilestone checker : milestones) {
+                if (checker.isOpen() && checker.isOngoing()) {
```

Comment:
From the way it's used, `isOngoing` only refers to open milestones. How about adding `isOpen` as part of `isOngoing` instead? I don't think any closed milestone will be ongoing. We can then `stream` the milestones and `findFirst`.